### PR TITLE
Task/geospatial tests

### LINKED ISSUE #3821: Add tests for api/app/utils/geospatial.py
**Describe the task**
Implement tests for test coverage for `raster_mul` and `warp_to_match_extent`.

**Acceptance Criteria**
- [ ] api/app/utils/geospatial.py is tested, ideally with 80% test coverage

**Additional context**
- Add any other context about the task here.
- Or here

### PR DESCRIPTION
**Pull request by conbrad**
Test suite for geospatial functions introduced in https://github.com/bcgov/wps/pull/3800

Closes #3821 
# Test Links:
[Landing Page](https://wps-pr-3841-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-3841-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-3841-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-3841-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3841-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3841-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3841-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3841-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: api/app/utils/geospatial.py [hunk 1]
```diff
@@ -1,66 +1,64 @@
...
 from osgeo import gdal
 
 
 logger = logging.getLogger(__name__)
 
 
-def warp_to_match_extent(source_raster: gdal.Dataset, raster_to_match: gdal.Dataset, output_path: str) -> gdal.Dataset:
+def warp_to_match_extent(source_ds: gdal.Dataset, ds_to_match: gdal.Dataset, output_path: str) -> gdal.Dataset:
     """
     Warp the source_raster to match the extent and projection of the other raster.
```
Comment (dgboss) [inline api/app/utils/geospatial.py:10]:
```suggestion
    Warp the source dataset to match the extent and projection of the other dataset.
```

File: api/app/tests/utils/test_geospatial.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    tpi_ds: gdal.Dataset = get_tpi_raster_wrong_shape()
+
+    with pytest.raises(ValueError):
+        raster_mul(tpi_ds, hfi_ds)
+
+    hfi_ds = None
+    tpi_ds = None
+
+
+@pytest.mark.skip(reason="enable once gdal is updated past version 3.4")
```
Comment (dgboss) [inline api/app/tests/utils/test_geospatial.py:91]: Just curious why we have to wait?

**Review by dgboss — APPROVED** [review]

File: api/app/tests/utils/test_geospatial.py (hunk 2 above)
Comment (conbrad) [reply api/app/tests/utils/test_geospatial.py:91]: Works locally, fails with API image

**Review by brettedw — APPROVED** [review]
👌

**Pull request merged**